src/JestTestRunner.ts

```diff
@@ -1,131 +1,49 @@
 import { EventEmitter } from 'events';
-
-import * as _ from 'lodash';
-const os = require('os');
-const path = require('path');
-
-import * as log4js from 'log4js';
-const log = log4js.getLogger('JestTestRunner');
-
-import { findJestAdapter, JestAdapter } from './JestVersionAdapters';
-
-const DEFAULT_OPTIONS: Object = {
-  cacheDirectory: path.join(os.tmpdir(), 'jest'),
-  collectCoverage: false,
-  haste: {},
-  maxWorkers: 1,
-  moduleFileExtensions: ['js', 'json', 'jsx', 'node'],
-  moduleNameMapper: [],
-  setupFiles: [],
-  snapshotSerializers: [],
-  testEnvironment: 'jest-environment-jsdom',
-  testRegex: '(/__tests__/.*|(\\.|/)(test|spec))\\.jsx?$',
-  testRunner: 'jest-jasmine2',
-  verbose: true
-};
+import JestTestAdapterFactory from './jestTestAdapters/JestTestAdapterFactory';
 
 export default class JestTestRunner extends EventEmitter implements TestRunner {
-  // Seems there are no up-to-date TypeScript definitions for Jest
+  private jestConfig: any;
+  private projectRoot: string;
 
-  private jestAdapter: JestAdapter;
-  private hasteContext: Promise<any>;
-  private options: any;
-  private paths: string[];
-
-  constructor(options: RunnerOptions) {
+  public constructor(options: RunnerOptions) {
     super();
-    this.jestAdapter = findJestAdapter();
-    log.debug(`Received options ${JSON.stringify(options)}`);
-
-    this.options = _.assign(DEFAULT_OPTIONS, {
-      rootDir: process.cwd(),
-      roots: [process.cwd()],
-      testPathDirs: [process.cwd()]
-    });
-    log.debug(`Using options ${JSON.stringify(this.options)}`);
-    
-    const _testRegex = new RegExp(this.options.testRegex);
-    const isTest = (file: string) => _testRegex.test(file);
 
-    this.paths = options.files
-      .map(file => file.name)
-      .filter(isTest);
+    this.projectRoot = process.cwd();
 
-    log.info(`Discovered specs ${JSON.stringify(this.paths)}`);
+    this.jestConfig = JSON.parse(options.strykerOptions.jest.config);
+    this.jestConfig.rootDir = this.projectRoot;
   }
 
-  init(): Promise<any> | void {
-    log.info(`Initializing Jest`);
-    this.hasteContext = this.jestAdapter.buildHasteContext(this.options);
-  }
+  public async run(): Promise<RunResult> {
+    const jestTestRunner = JestTestAdapterFactory.getJestTestAdapter();
 
-  run(): Promise<RunResult> {
-    return this.hasteContext
-      .then((context: any) => this.runTests(context))
-      .then((results: any[]) => this.processJestResults(results))
-      .catch((error: Error) => this.catchError(error));
-  }
+    const { results } = await jestTestRunner.run(this.jestConfig, process.cwd());
 
-  private runTests(hasteContext: any): Promise<any[]> {
-    const promises = this.paths.map((specPath: string) => {
-      return this.jestAdapter.runTest(
-        path.resolve(specPath),
-        this.options,
-        hasteContext.resolver
-      );
-    });
-    return Promise.all(promises);
+    return this.processRunResults(results);
```

Comment:
Right now, you call a method that just returns an object. I would prefer just returning the object here

Reply:
So assign the output of the function to an extra variable and return that?